lib/spack/spack/fetch_strategy.py

```diff
@@ -1575,16 +1575,30 @@ def for_package_version(pkg, version):
 
     check_pkg_attributes(pkg)
 
-    if not isinstance(version, spack.version.Version):
+    if not isinstance(version, spack.version.VersionBase):
```

Comment:
```suggestion
    if not isinstance(version, (spack.version.VersionBase, spack.version.GitVersion)):
```

Reply:
GitVersion inherits from VersionBase

Reply:
Yup, that's why I resolved it and added the comment to that effect.